python/pylibcudf/pylibcudf/io/csv.pyx

```diff
@@ -263,3 +269,195 @@ def read_csv(
         c_result = move(cpp_read_csv(options))
 
     return TableWithMetadata.from_libcudf(c_result)
+
+
+# TODO: Implement the remaining methods
+cdef class CsvWriterOptions:
+    """The settings to use for ``write_csv``
+
+    For details, see :cpp:class:`cudf::io::csv_writer_options`
+    """
+    @staticmethod
+    def builder(SinkInfo sink, Table table):
+        """Create a CsvWriterOptionsBuilder object
+
+        For details, see :cpp:func:`cudf::io::csv_writer_options::builder`
+
+        Parameters
+        ----------
+        sink : SinkInfo
+            The sink used for writer output
+        table : Table
+            Table to be written to output
+
+        Returns
+        -------
+        CsvWriterOptionsBuilder
+            Builder to build CsvWriterOptions
+        """
+        cdef CsvWriterOptionsBuilder csv_builder = CsvWriterOptionsBuilder.__new__(
+            CsvWriterOptionsBuilder
+        )
+        csv_builder.c_obj = csv_writer_options.builder(sink.c_obj, table.view())
```

Comment:
This object needs to keep `table` alive. And then when you call build it needs to transfer `table` to the returned `CsvWriterOptions` object.

(deja vu: https://github.com/rapidsai/cudf/pull/17263#discussion_r1843670559)